Library/Homebrew/utils/curl.rb

```diff
@@ -19,14 +19,7 @@ module Curl
     def curl_executable(use_homebrew_curl: false)
       return Formula["curl"].opt_bin/"curl" if use_homebrew_curl
 
-      @curl ||= [
-        ENV["HOMEBREW_CURL"],
```

Comment:
Why didn't this work before? Doesn't this mean that `HOMEBREW_CURL` had priority?

Reply:
This worked within `brew`, but any formula invoking `curl` would use whatever was in the PATH.

`HOMEBREW_CURL` still has priority in the shim.

Reply:
By "formula invoking `curl`" do you mean during the install process or when the installed executable is being run? (I've not played a ton with the shim setup we have so I don't totally know how it works)

Reply:
The install process. The `install`, `postinstall` and `test` blocks.

Reply:
Ah, I think I understand. Just to make sure:

If a formula calls something like this in the `install`, `postinstall`, or `test` block:

```rb
system "curl", "foo"
```

It will just use the `curl` that's found in the `PATH`. _Now_, the first `curl` that will be found will always be the shim which will then choose the correct `curl` to use, right? Had a formula been using e.g. `curl_output` (not sure if that was possible), it wouldn't have a problem because within the `Utils::Curl` module, the correct `curl` was already being used.

Is that mostly correct?

Reply:
Yes that's completely correct.